tests/modules/asset_groups/test_models.py

```diff
@@ -108,3 +108,65 @@ def test_asset_group_sightings_bulk(
             )
 
         tus_utils.cleanup_tus_dir(transaction_id)
+
+
+def test_asset_group_sighting_get_completion(
+    flask_app, flask_app_client, researcher_1, test_root, request
+):
+    from app.modules.asset_groups.models import (
+        AssetGroupSighting,
+        AssetGroupSightingStage,
+    )
+
+    # Create asset group sighting
+    transaction_id, test_filename = asset_group_utils.create_bulk_tus_transaction(
+        test_root
+    )
+    data = asset_group_utils.get_bulk_creation_data(transaction_id, test_filename)
+    # Use a real detection model to trigger a request sent to Sage
+    data.set_field('speciesDetectionModel', ['african_terrestrial'])
+    # and the sim_sage util to catch it
+    resp = asset_group_utils.create_asset_group_sim_sage(
+        flask_app, flask_app_client, researcher_1, data.get()
+    )
+    asset_group_guid = resp.json['guid']
+    request.addfinalizer(
```

Comment:
is this a neater way to ensure that things are cleaned up than the try/finlly blocks I've been using?
If so, thanks, I'll use them

Reply:
limiting indentation is good.
But surely the asset group is only deleted at the end, not after it's created which is here?
Just checking my understanding

Reply:
Any functions given to `request.addfinalizer` is only executed after the test, it's clean up code.  Just make sure you don't call the function in `request.addfinalizer`.